scene/gui/button.cpp

```diff
@@ -225,14 +237,37 @@ void Button::_notification(int p_what) {
 				style2->draw(ci, Rect2(Point2(), size));
 			}
 
-			Ref<Texture2D> _icon;
-			if (icon.is_null() && has_theme_icon(SNAME("icon"))) {
+			drawable_size_remained -= style->get_minimum_size();
```

Comment:
You need to explicitly subtract margins. While it's the default value, `get_minimum_size` is not guaranteed to return a value equal to margin sizes. Because this method can be overridden.

Reply:
If so, the size of the space for drawing icons and text will be different from that in `Button::get_minimum_size`.

Why not make the get method of margin overridable. `get_minimum_size` can then be overridden indirectly. If it is directly covered, it may actually be drawn incorrectly when using margin to draw.

Reply:
> If so, the size of the space for drawing icons and text will be different from that in `Button::get_minimum_size`.

Well, drawing happens on the actual size, not on the minimum size. Button can expand beyond its minimum size and that shouldn't affect the drawing logic, so such a discrepancy should not lead to problems.

The problem is that the minimum size of a stylebox should be _at least_ equal to its margins. Which is why by default `StyleBox::get_minimum_size()` returns a sum of stylebox's margins, and a lot of GUI code just assumes that the method is a shorthand for getting all the margins combined. This is an incorrect assumption.

The minimum size can be changed for a particular stylebox implementation. And when you calculate the minimum size of the button, you should respect that and take the full `StyleBox::get_minimum_size()` into account. But when you're deciding on the drawing area, you should only offset it based on the margin values, and not on the min size. Perhaps it's worth adding a method to stylebox to return the combined value of margins, because we've already needed that in the recent Tree PR. But for now, you can simply explicitly make `get_margin` calls for each side and use that.